### PR TITLE
Documentation and LobbyGame message

### PR DESCRIPTION
**Pull request by C-Merrill**
I think I've finished all the messages for lobby/pregame in the documentation. It doesn't format very well, but the information is there. I also made it so GET GAMES will serialize and send a vector of lobbygames.
There was a fix I did in ServerNetworkConnection that fixed a write exception in some test cases, but it caused non-responsiveness when working with an actual connected client. Probably something to do with bool connected either not getting set to true or getting reset to false when the connection is still live. I commented out the fix to make it work, but that does mean regression in unit tests. @droidkfx can you look at it for me?

### CODE REVIEW

**Review by katiesweet — APPROVED** [review]
I think this looks good.

@C-Merrill  and @michaelkamerath: 
I have locally added a START option the client can call to start the game and fill it in with AI's. I've also added an error check so you can't join the same game multiple times, and I was going to add an option to leave a game in the lobby as well. Seems to work fine interacting with my client, but there are a few bad file descriptor warnings that pop up during run time.  I'll put a pull request in when it's ready to go.

No promises on this since it's a last minute addition I thought we needed. Also, due to the time crunch, I'll probably not add tests for it since I'm testing it alongside my client, and I have lots I need to get done today.

**Comment by C-Merrill** [comment]
@katiesweet all of that sounds good. Are you ready to make a pull request with it?

**Comment by katiesweet** [comment]
Yes and no @C-Merrill ... There is a lot of little things I'm trying to work out. I can push up what I currently have pushed to my repository so you can begin looking at it. I'll make a pull request with that. I can add separate changes in a different pull request if you want, or we can wait and I keep pushing changes to that same pull request. Your choice.

File: source/NetworkInterface/TCPConnection.cpp [hunk 1]
```diff
@@ -76,8 +76,8 @@ void TCPConnection::aSyncRead(std::function<void(std::string)> callback)
 
 void TCPConnection::write(std::string msg)
 {
-  if (connected)
-  {
+  /*if (connected)
+  {*/
```
Comment (katiesweet) [inline source/NetworkInterface/TCPConnection.cpp:80]: In my pull request, I kept these commented in but had the ServerNetworkInterface set this to be true when someone connects successfully.

**Comment by C-Merrill** [comment]
I'm pulling this in. We don't have the time to wait.

**Pull request merged**